### ISSUE TITLE
restructuring: true is ruing the css 

### ISSUE THREAD

**Issue by ezraroi** [jakubpawlowicz/clean-css#598]
Since the last versions, the restructuring of the css changes the way things are displayed on the screen. A good example for this is the [UI-Grid](https://github.com/angular-ui/ng-grid). When using the css after minify things looks different without the minification (started in version ~3).

**Comment by jakubpawlowicz** [comment]
Would you be able to point me to a more distilled test case? Does it affect all browsers, etc?

**Comment by ezraroi** [comment]
Yes, all web browsers.

Will try to create a plnker with the 2 versions to demonstrate the problem

**Comment by jakubpawlowicz** [comment]
Would be great! :+1:

**Comment by jakubpawlowicz** [comment]
@ezraroi any updates?

**Comment by ezraroi** [comment]
sorry, I had few private issues in the last 2 weeks. Will update soon I hope

**Comment by jakubpawlowicz** [comment]
Sure, no sweat, just wanted to make sure it's still an issue.

**Comment by ezraroi** [comment]
yes, still an issue, broke my project until i turned it off

**Comment by jakubpawlowicz** [comment]
Sorry to hear, but let's fix it.

**Comment by ezraroi** [comment]
Original CSS Plunker: http://plnkr.co/edit/7lwoNsLZkHGZnJBsJ6fQ?p=preview
Minifed CSS Plunker: http://plnkr.co/edit/EtylR4dw5TUuZiWuuARx?p=preview

You can see that in the minifed version there are black borders between columns.

This how I minifed the css file:

```javascript
var fs = require('fs');
var CleanCSS = require('./node_modules/clean-css');
var css = fs.readFileSync("ngGrid.css", encoding='utf8');
var min = new CleanCSS().minify(css).styles;
fs.writeFileSync("ngGrid.min.css", min, {encoding : 'utf8'});
```

**Comment by jakubpawlowicz** [comment]
@ezraroi thanks for sharing, I will give it a try and get back to you.

**Comment by ezraroi** [comment]
@jakubpawlowicz np

**Comment by xak2000** [comment]
The problem is rule:
```
.ui-grid-cell {
  overflow: hidden;
  float: left;
  background-color: inherit;
  border-right: 1px solid;
  border-color: #d4d4d4;
  box-sizing: border-box;
}

.ui-grid-footer-cell {
  overflow: hidden;
  background-color: inherit;
  border-right: 1px solid;
  border-color: #d4d4d4;
  box-sizing: border-box;
  display: table-cell;
}  
```

becomes:

```
.ui-grid-cell,.ui-grid-footer-cell{overflow:hidden;border-color:#d4d4d4}
.ui-grid-cell{float:left;background-color:inherit;border-right:1px solid;box-sizing:border-box}
```

Interesting things here is united rule for `border-color:#d4d4d4` and `border-right:1px solid;` rule.

In first case (before minification) browser interprets `border-right:1px solid;` as:
```
    border-right-width: 1px;
    border-right-style: solid;
    border-right-color: rgb(212, 212, 212);
```

In second case (after minification) browser interprets `border-right:1px solid;` as:
```
    border-right-width: 1px;
    border-right-style: solid;
    border-right-color: initial;
```

I thing the problem is order of `border-*` rules.
In original `ui-grid.css` the rule `border-color:#d4d4d4` is **after** `border-right: 1px solid;` rule but in minified css it appears **before** it.

**Comment by jakubpawlowicz** [comment]
@xak2000 Please use the latest 3.3 or 3.4 as it's fixed there.

**Comment by xak2000** [comment]
I am using `"version": "3.4.1"`. Not fixed.

**Comment by xak2000** [comment]
You can download [ui-grid.css](https://cdn.rawgit.com/angular-ui/bower-ui-grid/master/ui-grid.css) and run `cleancss ./ui-grid.css -b | grep .ui-grid-cell[,{]` and you see 4th and 5th lines.

**Comment by jakubpawlowicz** [comment]
This is the correct test case:

```
.ui-grid-cell {
  overflow: hidden;
  border-right: 1px solid;
  border-color: #d4d4d4;
}
.ui-grid-cell:last-child {
  border-right: 0;
}
.ui-grid-footer-cell {
  overflow: hidden;
  border-right: 1px solid;
  border-color: #d4d4d4;
}
```

Fix is coming.

**Comment by xak2000** [comment]
Sorry. Didn't know that `.ui-grid-cell:last-child` affect this. Tried only with full file.
Thanks for fixing!

**Comment by jakubpawlowicz** [comment]
@xak2000 no worries, fix is ready on 3.4 branch, to be released tomorrow

**Comment by jakubpawlowicz** [comment]
It's released as 3.4.2.

**Comment by jakubpawlowicz** [comment]
Thanks for helping out with narrowing it down @ezraroi and @xak2000!
